### PR TITLE
Add io.github.mak448a.QTCord

### PR DESCRIPTION
**Pull request by mak448a**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by mak448a** [comment]
By the way, the icon isn't showing up in the menu on my KDE install. Can anyone please help?

**Comment by mak448a** [comment]
Ooops I still need to add a few more files. Didn't read the instructions fully.

File: io.github.mak448a.QTCord.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - mkdir -p /app/share/applications/
+      - install -Dm 644 src/smiley.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - mv src/io.github.mak448a.QTCord.desktop /app/share/applications/io.github.mak448a.QTCord.desktop
+      - mv src/* /app/bin
+    sources:
+      - type: git
+        url: https://github.com/mak448a/QTCord
+        tag: 0.0.0
+        commit: 05b112b22d3d7fa6daea79a2c2666ff403795130
+finish-args:
```
Comment (hfiguiere) [inline io.github.mak448a.QTCord.yml:21]: finish-args is usually above modules.

Comment (mak448a) [reply io.github.mak448a.QTCord.yml:22]: Thanks!

File: io.github.mak448a.QTCord.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  --socket=wayland
+  --share=network
```
Comment (hfiguiere) [inline io.github.mak448a.QTCord.yml]: missing `--socket=fallback-x11`, and `--share=ipc`

Comment (mak448a) [reply io.github.mak448a.QTCord.yml]: Ok adding

File: io.github.mak448a.QTCord.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+command: io.github.mak448a.QTCord
+modules:
+  - python3-modules.json
+  - name: BuildQTCord
+    buildsystem: simple
+    build-commands:
+      - install -D src/flatpak_builder/io.github.mak448a.QTCord /app/bin/io.github.mak448a.QTCord
+      - mkdir -p /app/share/applications/
+      - install -Dm 644 src/smiley.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - mv src/io.github.mak448a.QTCord.desktop /app/share/applications/io.github.mak448a.QTCord.desktop
```
Comment (hfiguiere) [inline io.github.mak448a.QTCord.yml]:
```suggestion
      - install -Dm 644 src/smiley.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
      - intsall -Dm644 src/io.github.mak448a.QTCord.desktop /app/share/applications/io.github.mak448a.QTCord.desktop
```

Comment (mak448a) [reply io.github.mak448a.QTCord.yml]: I'll add that

**Comment by hfiguiere** [comment]
bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
Why does it say `finishargs not defined`?

File: io.github.mak448a.QTCord.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+app-id: io.github.mak448a.QTCord
+runtime: org.kde.Platform
+runtime-version: '5.15-23.08'
+sdk: org.kde.Sdk
+command: io.github.mak448a.QTCord
+finish-args:
+  --socket=wayland
+  --share=network
+  --share=ipc
+  --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.mak448a.QTCord.yml:10]:
```suggestion
finish-args:
  - --socket=wayland
  - --share=network
  - --share=ipc
  - --socket=fallback-x11
```

**Comment by mak448a** [comment]
I tried running the linter on my system but it didn't work. Any idea how to fix this?

File: io.github.mak448a.QTCord.yml (hunk 4 above)
Comment (mak448a) [reply io.github.mak448a.QTCord.yml:10]: I'll try that.

**Comment by mak448a** [comment]
bot, build io.github.mak448a.QTCord

**Comment by hfiguiere** [comment]
> Why does it say `finish-args-not-defined`?

because it expects an array, which this was not.

**Comment by mak448a** [comment]
> > Why does it say `finish-args-not-defined`?
> 
> because it expects an array, which this was not.

I see, it wasn't proper yaml. Thank you

**Comment by mak448a** [comment]
Hm, looks like I need an xml file

**Comment by hfiguiere** [comment]
the appstream file is required.

**Comment by mak448a** [comment]
Is this the correct documentation? https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/

**Comment by mak448a** [comment]
bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
Feeling a bit lost here. I can't figure out how to get this to work.
bot, build io.github.mak448a.QTCord

File: io.github.mak448a.QTCord.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 mak448a -->
+<component type="desktop-application">
+    <id>io.github.mak448a.QTCord</id>
+    <launchable type="desktop-id">io.github.mak448a.QTCord.desktop</launchable>
```
[the remark this answers is not in the document]
Comment (mak448a) [reply io.github.mak448a.QTCord.metainfo.xml:5]: Ok

**Comment by mak448a** [comment]
Ok, I hope this works!
bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
Hm, it's giving me these errors:
```json
{
    "errors": [
        "appstream-screenshots-not-mirrored-in-ostree",
        "appstream-failed-validation"
    ]
}
```

**Comment by mak448a** [comment]
Hm, it says: 
```json
{
    "errors": [
        "appstream-screenshots-not-mirrored-in-ostree",
        "appstream-failed-validation"
    ]
}```

**Comment by bbhtt** [comment]
> Build [87892 failed](https://buildbot.flathub.org/#/builders/6/builds/87892)

You are missing release, validate the final appdata https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/#use-flathubs-appstream-util

**Comment by mak448a** [comment]
Is this better @bbhtt?

**Comment by mak448a** [comment]
bot, build [io.github.mak448a.QTCord](https://github.com/mak448a/flathub/tree/io.github.mak448a.QTCord)

**Comment by mak448a** [comment]
bot, build https://github.com/mak448a/flathub/tree/io.github.mak448a.QTCord

**Comment by mak448a** [comment]
bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
bot, build io.github.mak448a.QTCord

**Comment by bbhtt** [comment]
It'd be nice, if you can include the appdata in your upstream repository. It can be used with any packaging format, and you want to manage translations, releases etc. through upstream.

**Comment by mak448a** [comment]
Oh ok, sure.

File: io.github.mak448a.QTCord.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+app-id: io.github.mak448a.QTCord
+runtime: org.kde.Platform
+runtime-version: '5.15-23.08'
+sdk: org.kde.Sdk
+command: io.github.mak448a.QTCord
+finish-args:
+  - --socket=wayland
+  - --share=network
+  - --share=ipc
+  - --socket=fallback-x11
```
Comment (bbhtt) [inline io.github.mak448a.QTCord.yml:10]: Add `--device=dri`


```
libEGL warning: wayland-egl: could not open /dev/dri/renderD128 (No such file or directory)
MESA: error: ZINK: failed to choose pdev
libEGL warning: egl: failed to create dri2 screen
```

Comment (mak448a) [reply io.github.mak448a.QTCord.yml:10]: Thanks

**Comment by bbhtt** [comment]
Why do you need this? https://github.com/mak448a/QTCord/blob/f2a17a39fafc87f1e6b807d882541a98b45562ba/src/flatpak_builder/io.github.mak448a.QTCord#L10-L12

Why not include pyside6 directly here?

**Comment by mak448a** [comment]
It wasn't working properly adding it with flatpak-pip-generator, so I decided to pull it at run time.

**Comment by mak448a** [comment]
I'll try adding it manually

File: io.github.mak448a.QTCord.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+app-id: io.github.mak448a.QTCord
+runtime: org.kde.Platform
+runtime-version: '5.15-23.08'
```
Comment (bbhtt) [inline io.github.mak448a.QTCord.yml:3]: I'm pretty sure this should be 6.6, since you are using pyside 6.6.1 which is Qt 6.6.1

**Comment by bbhtt** [comment]
> It wasn't working properly adding it with flatpak-pip-generator, so I decided to pull it at run time.

It's because pyside6 doesn't publish a source package yet has arch specific wheels.

**Comment by mak448a** [comment]
How do I add wheel files to the json file that it generates?

**Comment by bbhtt** [comment]
Yes, like that but include aarch64 wheels too, since you are building for that arch.

```
		{
			"type": "file",
			"url": "https://files.pythonhosted.org/packages/56/65/50d0ced768b717f709a5ab224b648eb533a862efd0cde67d19c2727200b0/PySide6_Essentials-6.6.1-cp38-abi3-manylinux_2_28_x86_64.whl",
			"sha256": "c7185616083eab6f42eaed598d97d49fac4f60ae2e7415194140d54f58c2b42c",
			"only-arches": ["x86-64"]
		},
		{
			"type": "file",
			"url": "https://files.pythonhosted.org/packages/73/28/1313720f3e30304b1fd13aab139ec21f30a01f237e70e631663ddad21fd1/PySide6_Essentials-6.6.1-cp38-abi3-manylinux_2_31_aarch64.whl",
			"sha256": "a383c3d60298392cfb621ec1a0cf24b4799321e6c5bbafc021d4cc8076ea1315",
			"only-arches": ["aarch64"]
		}
```

Then do this for pyside6, pyside6-essentials, addons, shiboken.

**Comment by mak448a** [comment]
Ok. Thank you! I'll work on this later.

**Comment by bbhtt** [comment]
Here, just add the aarch64 sources:

<details>
<summary> pyside6 </summary>

```json
{
    "name": "python3-pyside6",
    "buildsystem": "simple",
    "build-commands": [
        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"pyside6\" --no-build-isolation"
    ],
    "sources": [
        {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/24/3a/a970808004b16dabdfaf77fa602b43a85c4d8812709a8bae065577283c4c/PySide6-6.6.1-cp38-abi3-manylinux_2_28_x86_64.whl",
            "sha256": "0a67587c088cb80e90d4ce3023b02466ea858c93a6dc9c4e062b13314e03d464"
        },
        {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/ec/b8/1f5335580241c3863584173308c842ffc74e05074f3f72b49e5b54ca18e3/PySide6_Addons-6.6.1-cp38-abi3-manylinux_2_28_x86_64.whl",
            "sha256": "a0982da4033319667f9df5ed6fa8eff300a88216aec103a1fff6751a172b19a0"
        },
        {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/56/65/50d0ced768b717f709a5ab224b648eb533a862efd0cde67d19c2727200b0/PySide6_Essentials-6.6.1-cp38-abi3-manylinux_2_28_x86_64.whl",
            "sha256": "c7185616083eab6f42eaed598d97d49fac4f60ae2e7415194140d54f58c2b42c"
        },
        {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/bb/72/e54f758e49e8da0dcd9490d006c41a814b0e56898ce4ca054d60cdba97bd/shiboken6-6.6.1-cp38-abi3-manylinux_2_28_x86_64.whl",
            "sha256": "fb102e4bc210006f0cdd0ce38e1aaaaf792bd871f02a2b3f01d07922c5cf4c59"
        }
    ]
}

</details>

**Comment by mak448a** [comment]
> Here, just add the aarch64 sources:
> pyside6

Thanks! I added them and removed that part that installs pyside in the run script.

**Comment by bbhtt** [comment]
You're missing the aarch64 sources https://github.com/flathub/flathub/pull/4785#issuecomment-1857219325

**Comment by mak448a** [comment]
Oh right

**Comment by mak448a** [comment]
Ok I added them. bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
Oh wait I forgot something

File: io.github.mak448a.QTCord.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      - install -Dm 644 src/smiley.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - install -Dm644 src/io.github.mak448a.QTCord.desktop /app/share/applications/io.github.mak448a.QTCord.desktop
+      - mv src/* /app/bin
+    sources:
+      - type: git
+        url: https://github.com/mak448a/QTCord
+        tag: v0.0.1
+        commit: 0013b5ffe254b8d240b415d5f1480c49e632076b
+      - type: file
+        path: io.github.mak448a.QTCord.metainfo.xml
```
Comment (bbhtt) [inline io.github.mak448a.QTCord.yml:29]:
```suggestion
```

File: io.github.mak448a.QTCord.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  - --socket=fallback-x11
+  - --device=dri
+modules:
+  - python3-modules.json
+  - name: BuildQTCord
+    buildsystem: simple
+    build-commands:
+      - install -D src/flatpak_builder/io.github.mak448a.QTCord /app/bin/io.github.mak448a.QTCord
+      - mv licenses/ /app/bin/
+      - install -Dm 644 io.github.mak448a.QTCord.metainfo.xml /app/share/metainfo/io.github.mak448a.QTCord.metainfo.xml
```
Comment (bbhtt) [inline io.github.mak448a.QTCord.yml:19]: Swicth it to upstream https://github.com/mak448a/QTCord/blob/main/src/flatpak_builder/io.github.mak448a.QTCord.metainfo.xml

Comment (mak448a) [reply io.github.mak448a.QTCord.yml:19]: How do I include the file?
Like this?

```
sources:
  - type: file
        url: https://github.com/mak448a/QTCord/blob/main/src/flatpak_builder/io.github.mak448a.QTCord.metainfo.xml
```

File: python3-modules.json [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+            "name": "python3-requests",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"requests\" --no-build-isolation"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/64/62/428ef076be88fa93716b576e4a01f919d25968913e817077a386fcbe4f42/certifi-2023.11.17-py3-none-any.whl",
+                    "sha256": "e036ab49d5b79556f99cfc2d9320b34cfbe5be05c5871b51de9329f0603b0474",
```
Comment (bbhtt) [inline python3-modules.json:16]:
```suggestion
                    "sha256": "e036ab49d5b79556f99cfc2d9320b34cfbe5be05c5871b51de9329f0603b0474"
```

File: io.github.mak448a.QTCord.yml (hunk 9 above)
Comment (bbhtt) [reply io.github.mak448a.QTCord.yml:19]: `install -Dm 644 src/flatpak_builder/io.github.mak448a.QTCord.metainfo.xml /app/share/metainfo/io.github.mak448a.QTCord.metainfo.xml`

Comment (bbhtt) [reply io.github.mak448a.QTCord.yml:19]: You have to update that file first here https://github.com/mak448a/QTCord/blob/main/src/flatpak_builder/io.github.mak448a.QTCord.metainfo.xml and make a new tag.

Comment (mak448a) [reply io.github.mak448a.QTCord.yml:19]: Ah ok

**Comment by mak448a** [comment]
bot, build io.github.mak448a.QTCord

**Comment by bbhtt** [comment]
> Build [88037 successful](https://buildbot.flathub.org/#/builders/6/builds/88037) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/70704/io.github.mak448a.QTCord.flatpakref
> ```

Seems fine now.

**Comment by barthalion** [comment]
bot, build io.github.mak448a.QTCord

**Comment by mak448a** [comment]
> yo um, on your github you mention how your account has been flagged twice from using this client, maybe you should have mentioned that in your appdata that shows on the main flathub website, otherwise it looks cool

Thanks! I'll add that

**Comment by mak448a** [comment]
How do I fix this error? https://github.com/flathub/io.github.mak448a.QTCord/pull/1

**Pull request closed without merging**